_rules/image-no-text-0va7u6.md

```diff
@@ -0,0 +1,209 @@
+- the image resource contains text and it is [essential][] that the text is rendered with that specific presentation; or
+- the test target is [purely decorative][].
+
+## Assumptions
+
+- The text in the image resource is visible and not hidden by other resources rendered on the web page. If the text is not visible, even though parts of the image are, this rule may fail while [SC 1.4.5 Images of Text][sc1.4.5] and [SC 1.4.9 Images of Text (No Exception)][sc1.4.9] might by satisfied.
+- The page does not provide a mechanism to change the rendered text in the image resource. Otherwise, the rule might fail while [SC 1.4.5 Images of Text][sc1.4.5] and [SC 1.4.9 Images of Text (No Exception)][sc1.4.9] might by satisfied.
+- The specific presentation of the text rendered in the image resource can be achieved through formatted text. Otherwise, the rule might fail while [SC 1.4.5 Images of Text][sc1.4.5] and [SC 1.4.9 Images of Text (No Exception)][sc1.4.9] might by satisfied.
+
+## Accessibility Support
+
+_No accessibility support issues known._
+
+## Background
+
+This rule is designed specifically for [SC 1.4.5 Images of Text][sc1.4.5] which includes exceptions to the images it applies to. [SC 1.4.9 Images of Text (No Exception)][sc1.4.9] does not consider any exceptions. Therefore, some images that are inapplicable for this rule can be applicable to [SC 1.4.9 Images of Text (No Exception)][sc1.4.9].
+
+- [Understanding Success Criterion 1.4.5: Images of Text][sc1.4.5]
+- [Understanding Success Criterion 1.4.9: Images of Text (No Exception)][sc1.4.9]
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `img` element references an image resource that does not contain text.
+
+```html
+<img src="/test-assets/shared/fireworks.jpg" alt="fireworks going off behind the Eiffel tower at night" />
+```
+
+#### Passed Example 2
+
+This `input` element references an image resource that does not contain text.
+
+```html
+<input type="image" src="test-assets/shared/file.svg" alt="New file" />
+```
+
+#### Passed Example 3
+
+This `svg` element displays an image resource that does not contain text.
+
+```html
+<svg width="2in" height="3in" xmlns="http://www.w3.org/2000/svg">
+	<image x="20" y="20" width="200px" height="100px" href="/test-assets/shared/fireworks.jpg">
+		<title>Fireworks in Paris</title>
+	</image>
+</svg>
+```
+
+#### Passed Example 4
+
+This `object` element references an image resource where text is not the most significant content.
+
+```html
+<object data="/test-assets/0va7u6/times_square.jpg"></object>
+```
+
+#### Passed Example 5
+
+This `img` element references an image resource that contains text but where the presentation of the text is essential to convey the information.
+
+```html
+<p>The following image illustrates the use of cursive writing in the late nineteenth century.</p>
+
+<canvas id="canvas"> Your browser does not support the HTML5 canvas element. </canvas>
```

Comment:
I may be contradicting my previous review here, but I think we should exclude canvas. It can be really difficult, if not practically impossible to determine the source of a canvas, and in addition if you do want customizable images of text, canvas is the way to do it.

Reply:
In the "embedded image" definition I'm considering only canvas images that are rendered using the drawImage method. I can add an inapplicable example of using canvas with other rendering methods. Or would you still prefer to remove canvas from the applicability?

Reply:
How is someone supposed to know if drawImage was used? There is no API you can call to figure that out. That doesn't seem possible.